### ISSUE TITLE
Getting involved with KVision development

### ISSUE THREAD

**Issue by robert-cronin** [rjaros/kvision#36]
I've been developing with KVision for a couple of months so I can see the applicability as a web development tool. I'd love to get more involved. 

Are you open to getting some help in developing the platform? If so,  perhaps you could share some of the items on the to-do list you mentioned in another issue

**Comment by rjaros** [comment]
Of course I am open to cooperation :-) There is a lot of work to be done.

1. Documentation
- improvements to the guide (more examples, more options covered, language corrections)
- a tutorial for Kotlin beginners (targeted to JS developers)
- better code documentation (more accurate classes and methods descriptions) for better API doc
- interactive documentation (some kind of playground with live examples mentioned in #33)
- documentation for dynamic forms (#18) and dynamic tabs (#26)
- a dedicated webpage

2. Builder scripts for example applications
- webpack optimizations (dynamic imports are currently blocked by https://youtrack.jetbrains.com/issue/KT-28523)
- Gradle scripts in Kotlin instead of Groovy
 
3. Client side
- GraphQL client component
- integrated support for Cordova (or any similar framework targeting mobile platforms) with a template application and dedicated components/API

4. Server side interfaces
- tabulator-remote module (integrating tabulator tabs with remote data sources, similar to current select-remote module)
- support for Micronaut
- support for Spring Webflux (Webflux will soon support Kotlin coroutines, and current Spring Boot module uses MVC and runBlocking, which is not a perfect solution).
- serverless integration (AWS, Azure, Google - for now it's just an alluring idea)

**Comment by rjaros** [comment]
Of course it's also possible to just create new modules for many different popular JS components - e.g. CKEditor, Select2 or others.

**Comment by robert-cronin** [comment]
I think I might stick with either documentation expansion or adding a new JS module for now just for something to start with

**Comment by rjaros** [comment]
I have sent you an invite link to Gitbook KVision account.

**Comment by rjaros** [comment]
There are also a few bugs in other projects, that have impact on KVision development. Vote for them, please.
- https://youtrack.jetbrains.com/issue/KT-28523
- https://youtrack.jetbrains.com/issue/KT-27855
- https://github.com/Kotlin/kotlinx.serialization/issues/430
- https://github.com/ktorio/ktor/issues/926
- https://github.com/Kotlin/kotlinx.serialization/issues/113

**Comment by robert-cronin** [comment]
Got the invite and voted up those issues
